cpu/lpc2387/periph/rtc.c

```diff
@@ -56,10 +53,10 @@ void rtc_init(void)
 
     RTC_CCR = CCR_CLKSRC;                   /* Clock from external 32 kHz Osc. */
 
-    /* initialize clock with valid unix compatible values
-     * If RTC_YEAR contains an value larger unix time_t we must reset. */
+    /* initialize clock with valid unix compatible values */
```

Comment:
I think that the year 2038 issue is solved in most UNIX compatible systems by now. So if there is some merit in bug-compatibility with other UNIX-like systems (which I doubt), this bug-compatibility is no longer needed.

What would make more sense to me is to set the time to some predictable and reasonable default (e.g. 1st January 1970 0:00:00 seems to be a good candidate for that default) on a cold boot. Doing the reset only depending on what random state the RTC has after a cold boot seems strange to me.

Reply:
It involved some trickery because I didn't want to waste a precious byte of RAM for that, but by clearing the POR bit after Deep Sleep we can distinguish between Deep Sleep and (real) POR and only reset the RTC in the later case.